### ISSUE TITLE
remove_snps() - Error in h(simpleError(msg, call))

### ISSUE THREAD

**Issue by YoannPa** [CompEpigen/methrix#25]
After loading a methrix object from a RDS file ( readRDS() ), when I tried to remove SNPs this error message shown up (partially translated from French):
```
Error in h(simpleError(msg, call)) : 
  evaluation error of the argument 'x' when selecting a method for 'unique' function: evaluation error of the argument 'x' when selecting a method for 'unique' function: evaluation error of the argument 'x' when selecting a method for 'which' function : Sequence names M in 'ranges' not present in reference genome hs37d5.

```
For some reason the error for unique() is printed 2 times. The 3 errors are concatenated (without carriage returns). The last part of the error message sounds like an attempt to query M chromosome on the annotation MafDb.1Kgenomes.phase3.hs37d5.

Here is the complete code:
```
#Load library
library(methrix)
library(BSgenome.Hsapiens.UCSC.hg19)

#Example bedgraph files 2 cancer samples and 2 normal samples
bdg_files <- list.files(
  path = system.file('extdata', package = 'methrix'),
  pattern = "*bedGraph\\.gz$",
  full.names = TRUE
)

#Generate some sample annotation table
sample_anno <- data.frame(
  row.names = gsub(
    pattern = "\\.bedGraph\\.gz$",
    replacement = "",
    x = basename(bdg_files)
  ),
  Condition = c("cancer", 'cancer', "normal", "normal"),
  Pair = c("pair1", "pair2", "pair1", "pair2"),
  stringsAsFactors = FALSE
)

#First extract genome wide CpGs from the desired reference genome
hg19_cpgs <- methrix::extract_CPGs(ref_genome = "BSgenome.Hsapiens.UCSC.hg19")

meth <- readRDS(file = "methrix_object.RDS")

#Generate report
methrix::methrix_report(meth = meth, output_dir = "~/methrix_tuto/")

# #Remove uncovered locis
# meth = methrix::remove_uncovered(m = meth)

if(!require(MafDb.1Kgenomes.phase3.hs37d5)) {
  BiocManager::install("MafDb.1Kgenomes.phase3.hs37d5")}
if(!require(GenomicScores)) {
  BiocManager::install("GenomicScores")}

library(MafDb.1Kgenomes.phase3.hs37d5)
library(GenomicScores)

#Remove SNPs
meth_snps_filtered <- methrix::remove_snps(m = meth)
```
The error only happens for the last command.  
I cannot reproduce the error when the methrix object is generated with read_bedgraphs(), instead of being loaded from a RDS file.

Let me know if you need anything else for the issue.
Best,

**Comment by tkik** [comment]
Hi, 

Can it happen that the RDS file was generated with an older version of methrix? In this case it might have chrM in it, which is not in the SNP database. Can you try to generate it with the current version? 
If you give me a few days, I can also implement a workaround in the remove_snp function. However, I think it is best not to keep the chrM in the methrix object, except if you have a specific reason, e.g. you are interested in chrM methylation. 
Best,

**Comment by YoannPa** [comment]
Okay I will try to update methrix to see if it changes something.
I keep you in touch !

**Comment by QianhuiWan** [comment]
Hello, I also have this issue when using my own WGBS data for remove_snps() function, is there a better way to remove chrM form meth object? Thank you so much. Best regards, Qianhui

**Comment by PoisonAlien** [comment]
Hi,
You could restrict your `methrix` object to only main contigs with `subset_methrix` function.

```r
> main_contigs = paste0("chr", c(1:22, "X", "Y")) #Assuming you are working with human samples
> main_contigs
 [1] "chr1"  "chr2"  "chr3"  "chr4"  "chr5"  "chr6"  "chr7"  "chr8"  "chr9"  "chr10" "chr11" "chr12" "chr13"
[14] "chr14" "chr15" "chr16" "chr17" "chr18" "chr19" "chr20" "chr21" "chr22" "chrX"  "chrY"
> meth_main = methrix::subset_methrix(m = meth, contigs = main_contigs)
```

Resulting `meth_main` should be devoid of chrM or any other unknown short contigs.
Hope this helps.

**Comment by QianhuiWan** [comment]
I tried `subset_methrix` and the problem was solved, thank you for your help!

**Comment by tkik** [comment]
Hi, 
the `subset_methrix` solution is a valid approach. But to make it easier in the future, I took a look and I am implementing some changes so it will be able to handle the chrMT in the `remove_snps`  function. I am testing it now and I will commit my changes in a few days, so please check back a bit later.

**Comment by tkik** [comment]
I fixed the issue in 6cd7fcf8bbebe6f0f60a63ea4e91fd81021e79c8.
